### PR TITLE
Add python_abi for 3.6 branch

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
Azure failure is expected. Appveyor passes.

**Comment by jjhelmus** [comment]
LGTM, thanks @isuruf

File: .ci_support/osx_python3.7.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  sqlite:
+    max_pin: x
+  tk:
+    max_pin: x.x
+  xz:
+    max_pin: x.x
+  zlib:
+    max_pin: x.x
+python:
+- '3.7'
```
Comment (jakirkham) [inline .ci_support/osx_python3.7.yaml]: This says `3.7`, but we are building `3.6` here. Is that expected?

Comment (isuruf) [reply .ci_support/osx_python3.7.yaml]: Yeah, that's fine. `conda-smithy` is confused here, but it doesn't matter.

**Pull request merged**